torch/csrc/jit/codegen/cuda/executor_kernel_arg.h

```diff
@@ -170,12 +319,60 @@ class KernelArgumentHolder {
 
   void push(const std::vector<at::Tensor>& tensors);
 
+  void push(const ArgAbstract* arg);
+
+  void swap(int i, const ArgAbstract* arg);
+
+  // push int64
+  void push(int64_t val);
+
+  const ArgAbstract* back() const {
+    return arguments_.back().get();
+  }
+
   void appendPhiloxRNGSeed(uint64_t rand_offset);
 
+  const ArgAbstract* operator[](int ind) const {
+    return arguments_.at(ind).get();
+  };
+
+  size_t size() const {
+    return arguments_.size();
+  }
+
+  bool empty() const {
+    return arguments_.empty();
+  }
+
+  void setDeviceIndex(int index) {
+    device_index_ = index;
+  }
+
+  int getDeviceIndex() const {
+    return device_index_;
+  }
+
+  void setCacheId(size_t id) {
+    cache_id_ = id;
+  }
+
+  c10::optional<size_t> getCacheId() const {
+    return cache_id_;
+  }
+
+  void print() const {
+    for (const auto& arg : arguments_) {
+      arg->print();
+    }
+  }
+
  private:
   std::vector<std::unique_ptr<ArgAbstract>> arguments_;
   std::vector<void*> void_ptrs_;
```

Comment:
`void_ptrs_` is used to hold the kernel arguments when we launch cuda kernels. `void_ptrs_.data()` is what gets returned.

The logic of keeping a `void_ptrs_` as a member is simply because these pointers are pointing to memory owned by KenerlArgumentHolder.